### PR TITLE
CAMEL-6132 - Initial version of camel-test-karaf

### PR DESCRIPTION
**Pull request by hqstevenson**
This PR provides the structure modifications.  The AbstractFeatureTest class used by camel-itest-karaf is now in the camel-test-karaf module, and has been enhanced to reduce the possibility of dangling Karaf instances.

### CODE REVIEW

File: components/camel-test-karaf/src/main/java/org/apache/camel/test/karaf/CamelKarafTestSupport.java [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+                keepRuntimeFolder(),
+                logLevel(LogLevelOption.LogLevel.INFO),
+                replaceConfigurationFile("etc/org.ops4j.pax.logging.cfg", getConfigFile("/etc/org.ops4j.pax.logging.cfg")),
+                editConfigurationFilePut("etc/system.properties", "hibernate3.version", System.getProperty("hibernate3.version")),
+                editConfigurationFilePut("etc/system.properties", "hibernate42.version", System.getProperty("hibernate42.version")),
+                editConfigurationFilePut("etc/system.properties", "hibernate43.version", System.getProperty("hibernate43.version")),
+                editConfigurationFilePut("etc/system.properties", "spring31.version", System.getProperty("spring31.version")),
+                editConfigurationFilePut("etc/system.properties", "spring32.version", System.getProperty("spring32.version")),
+                editConfigurationFilePut("etc/system.properties", "spring40.version", System.getProperty("spring40.version")),
+                editConfigurationFilePut("etc/system.properties", "spring41.version", System.getProperty("spring41.version"))
```
Comment (davsclaus) [inline components/camel-test-karaf/src/main/java/org/apache/camel/test/karaf/CamelKarafTestSupport.java]: We also have a spring 4.2 version

File: components/camel-test-karaf/src/main/java/org/apache/camel/test/karaf/CamelKarafTestSupport.java [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+        return new Option[]{
+                // KarafDistributionOption.debugConfiguration("8889", true),
+                karafDistributionConfiguration().frameworkUrl(maven().groupId("org.apache.karaf").artifactId("apache-karaf").versionAsInProject().type("tar.gz"))
+                        .karafVersion(MavenUtils.getArtifactVersion("org.apache.karaf", "apache-karaf")).name("Apache Karaf").unpackDirectory(new File("target/exam")),
+                keepRuntimeFolder(),
+                logLevel(LogLevelOption.LogLevel.INFO),
+                replaceConfigurationFile("etc/org.ops4j.pax.logging.cfg", getConfigFile("/etc/org.ops4j.pax.logging.cfg")),
+                editConfigurationFilePut("etc/system.properties", "hibernate3.version", System.getProperty("hibernate3.version")),
+                editConfigurationFilePut("etc/system.properties", "hibernate42.version", System.getProperty("hibernate42.version")),
+                editConfigurationFilePut("etc/system.properties", "hibernate43.version", System.getProperty("hibernate43.version")),
```
Comment (davsclaus) [inline components/camel-test-karaf/src/main/java/org/apache/camel/test/karaf/CamelKarafTestSupport.java]: Not sure we should have hibernate here, as Apache Camel has not hibernate component.

Comment (hqstevenson) [inline components/camel-test-karaf/src/main/java/org/apache/camel/test/karaf/CamelKarafTestSupport.java]: I shouldn't have put this class in yet.  I put it there to compare with the AbstractFeatureTest class to see if it made sense as a base class, but I'm not sure it does.  

My intention is to modify this class to become what we want for Camel.  The version I used with my customer didn't have any of the system.properties changes that are in this version.  But we should put in/take out whatever makes sense for Camel.

**Comment by davsclaus** [comment]
@hqstevenson do you want to continue working on this PR and we can merge later when its ready
Or do prefer we merge and you then need to do a new PR for new changes?

**Comment by hqstevenson** [comment]
@devclaus - I'd like to get this PR merged since it is the most invasive (pom changes and such).  Then I'll continue refining the CamelKarafTestSupport class, and probably add an example project for it.  Is that OK?

**Comment by davsclaus** [comment]
Yep sounds good. I have just merged this.

Can you close this PR

**Comment by hqstevenson** [comment]
Thank You

**Pull request closed without merging**